services/database/dune/fetchRows.ts

```diff
@@ -96,73 +102,85 @@ export async function fetchCompiledContracts(): Promise<
         runtime_code_hash,
         runtime_code_artifacts
       FROM compiled_contracts
-      LIMIT 1
+      ORDER BY created_at ASC
+      OFFSET ${page}
+      LIMIT ${pageSize}
     `;
   return fetchWithQuery("compiled_contracts", query);
 }
 
-export async function fetchCompiledContractsSources(): Promise<
-  PgCompiledContractsSource[] | null
-> {
-  return fetchWithQuery("compiled_contracts_sources");
+export async function fetchCompiledContractsSources(
+  page: number,
+  pageSize: number,
+): Promise<PgCompiledContractsSource[] | null> {
+  return fetchWithQuery(
+    "compiled_contracts_sources",
+    `SELECT * FROM compiled_contracts_sources ORDER BY id ASC OFFSET ${page} LIMIT ${pageSize}`,
+  );
 }
 
-export async function fetchContractDeployments(): Promise<
-  PgContractDeployment[] | null
-> {
-  return fetchWithQuery("contract_deployments");
+export async function fetchContractDeployments(
+  page: number,
+  pageSize: number,
+): Promise<PgContractDeployment[] | null> {
+  return fetchWithQuery(
+    "contract_deployments",
+    `SELECT * FROM contract_deployments ORDER BY created_at ASC OFFSET ${page} LIMIT ${pageSize}`,
+  );
 }
 
-export async function fetchContracts(): Promise<PgContract[] | null> {
-  return fetchWithQuery("contracts");
+export async function fetchContracts(
+  page: number,
+  pageSize: number,
+): Promise<PgContract[] | null> {
+  return fetchWithQuery(
+    "contracts",
+    `SELECT * FROM contracts ORDER BY created_at ASC OFFSET ${page} LIMIT ${pageSize}`,
+  );
 }
 
-export async function fetchSources(): Promise<PgSource[] | null> {
-  return fetchWithQuery("sources");
+export async function fetchSources(
+  page: number,
+  pageSize: number,
+): Promise<PgSource[] | null> {
+  return fetchWithQuery(
+    "sources",
+    `SELECT * FROM sources ORDER BY created_at ASC OFFSET ${page} LIMIT ${pageSize}`,
+  );
 }
 
-export async function fetchSourcifyMatches(): Promise<
-  PgSourcifyMatch[] | null
-> {
-  return fetchWithQuery("sourcify_matches");
+export async function fetchSourcifyMatches(
+  page: number,
+  pageSize: number,
+): Promise<PgSourcifyMatch[] | null> {
+  return fetchWithQuery(
+    "sourcify_matches",
+    `SELECT * FROM sourcify_matches ORDER BY created_at ASC OFFSET ${page} LIMIT ${pageSize}`,
```

Comment:
I wonder why you don't paginate by using `where id > ${last_id} limit ${pageSize}` 
Using offset is generally slow with high offsets because the database has to go over all rows smaller than the offset. I think it would be better to use the `id` based approach also here.

Regarding ordering, also the queries can be very slow because `created_at` is not indexed. I think it would be better to use `order by id` too.

This counts for all other queries too. Using indexed columns for iterating makes everything much more performent.

Reply:
The problem is that not all tables have an incremental id

Reply:
Does that matter? you can even do it on the code table where we have hashes as id.

Just tried this successfully for example:
```
where code_hash > decode('52E56732A80EA66B14A846E383A99CB3455871F2C899E23CE81F02EAA5F94C18', 'hex') order by code_hash limit 10;
```